xrpl/utils/tx_parser/balance_changes.py

```diff
@@ -0,0 +1,332 @@
+"""
+Parse balance changes, final balances and previous balances of every
+account involved in the given transaction.
+"""
+from __future__ import annotations
+
+from decimal import Decimal
+from typing import Any, Dict, List, Optional, Union
+
+from xrpl.utils.tx_parser.utils import (
+    compute_balance_changes,
+    is_valid_metadata,
+    normalize_metadata,
+    normalize_nodes,
+    parse_final_balance,
+    parse_quantities,
+)
+
+
+class BalanceChanges:
```

Comment:
Why are these all classes instead of methods? IMO if they're only existing to process the metadata, they should be methods.

Reply:
That way I think it is easier to access e.g. `.source_account_balances` directly from the class. If you think this is not necessary or too confusing I can easily change them up to functions and just return all balance changes, final balances, ... :)